src/common.jl

```diff
@@ -120,3 +120,14 @@ function calcscattermat(Z::DenseMatrix)
     return calcscattermat(SimpleCovariance(), Z)
 end
 
+
+# distance
+"""
+    L2distance(X)
+
+Calculate a symmetric Euclidean (L2) distance matrix.
+"""
+L2distance(X::AbstractMatrix{T}) where {T<:Real} = L2distance!(zeros(T,size(X,2),size(X,2)), X)
```

Comment:
Presumably we could use `Matrix{T}(undef, size(X, 2), size(X, 3))` but the performance gain is unlikely to be observable in practice.